homeassistant/components/shelly/climate.py

```diff
@@ -0,0 +1,190 @@
+        self._attr_unique_id = f"{self.wrapper.mac}-climate"
+        self._attr_preset_modes: list[str] = wrapper.device.settings[
+            "schedule_profile_names"
+        ]
+        self._attr_preset_modes.insert(0, PRESET_NONE)
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Set target temperature."""
+        return cast(float, self.block.targetTemp)
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return current temperature."""
+        return cast(float, self.block.temp)
+
+    # @property
+    # def is_available(self) -> bool:
+    #     """Device a vailability."""
+    #     return not cast(bool, self.block.valveError)
+
+    @property
+    def hvac_mode(self) -> str:
+        """HVAC current mode."""
+        if not hasattr(self.block, "mode") or self._check_is_off():
+            return HVAC_MODE_OFF
+
+        if cast(int, self.block.mode) != 0:
+            return HVAC_MODE_AUTO
+        return HVAC_MODE_HEAT
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Preset current mode."""
+        if not hasattr(self.block, "mode"):
+            return None
+        return self._attr_preset_modes[cast(int, self.block.mode)]
+
+    @property
+    def hvac_action(self) -> str | None:
+        """HVAC current action."""
+        if self._check_is_off():
+            return CURRENT_HVAC_OFF
+        return CURRENT_HVAC_HEAT
+
+    def _check_is_off(self) -> bool:
+        """Return if valve is off or on."""
+        return bool(
+            self.target_temperature is None
+            or (self.target_temperature <= self._attr_min_temp)
+        )
+
+    async def set_state_full_path(self, path: str, **kwargs: Any) -> Any:
+        """Set block state (HTTP request)."""
+        _LOGGER.debug("Setting state for entity %s, state: %s", self.name, kwargs)
+        try:
+            async with async_timeout.timeout(AIOSHELLY_DEVICE_TIMEOUT_SEC):
+                return await self.block.set_state_full_path(path, **kwargs)
+        except (asyncio.TimeoutError, OSError) as err:
+            _LOGGER.error(
+                "Setting state for entity %s failed, state: %s, error: %s",
+                self.name,
+                kwargs,
+                repr(err),
+            )
+            self.wrapper.last_update_success = False
+            return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        if (current_temp := kwargs.get(ATTR_TEMPERATURE)) is None:
+            return
+        await self.set_state_full_path(
+            "settings", target_t_enabled=1, target_t=f"{current_temp}"
+        )
+        if self._check_is_off():
+            self.async_set_hvac_mode(HVAC_MODE_OFF)
+        else:
+            self.async_set_hvac_mode(
+                HVAC_MODE_AUTO if self.preset_mode else HVAC_MODE_HEAT
+            )
+        self.async_write_ha_state()
```

Comment:
Why do we need to call write state, are we not getting a state pushed back that will trigger the state to be updated?

Reply:
Or does sending RPC commands already return and update the state in aioshelly?

Reply:
Leftover, removed